### PR TITLE
Progressive Wasserstein Barycenters of Persistence Diagrams

### PR DESCRIPTION
**Pull request by julesvidal**
Given a set of persistence diagrams, this plugin enables the computation of their Wasserstein Barycenter (the persistence diagrams which is the most representative of the set). The algorithm is progressive and interruptible. This method is extended to the computation of a clustering of the diagrams, in a K-Means-like approach.
A fast Auction-based implementation is also included for pairwise distances between diagrams.

### CODE REVIEW

File: core/vtk/ttkPersistenceDiagramClustering/ttkPersistenceDiagramClustering.h [hunk 1]
```diff
@@ -0,0 +1,1021 @@
+#ifndef _TTK_PERSISTENCEDIAGRAMSCLUSTERING_H
```
Comment (julien-tierny) [inline core/vtk/ttkPersistenceDiagramClustering/ttkPersistenceDiagramClustering.h]: Thanks for providing doxygen entries for authorship and references to the related publication. See core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.h for an example.
Thanks for reporting this also to the headers of the base layer.

File: core/base/persistenceDiagramClustering/PersistenceDiagramClustering.h [hunk 2]
```diff
@@ -0,0 +1,411 @@
+#ifndef _PERSISTENCEDIAGRAMSCLUSTERING_H
```
Comment (julien-tierny) [inline core/base/persistenceDiagramClustering/PersistenceDiagramClustering.h]: Thanks for adding authorship and references.

**Comment by julien-tierny** [comment]
awesome. thanks a lot.
thanks for considering my remarks in the code review.
also thanks for checking that the build has no warning.
cheers,
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

File: core/base/persistenceDiagramClustering/PersistenceDiagramClustering.h (hunk 2 above)
Comment (julesvidal) [reply core/base/persistenceDiagramClustering/PersistenceDiagramClustering.h]: All done! Also there shouldn't be any warnings anymore.
cheers,
Jules

**Comment by julien-tierny** [comment]
perfect, thanks!
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

**Pull request merged**